app/commands/solution/create.rb

```diff
@@ -9,6 +9,7 @@ def call
       begin
         solution_class.create!(user: user, exercise: exercise).tap do |solution|
           record_activity!(solution)
+          AwardBadgeJob.perform_later(user, :new_years_resolution)
```

Comment:
I've currently awarded the badge when a solution is created, not when it was completed. Let me know if that is okay.